### PR TITLE
CORE-12019 Add a Fast Check for already running CSDE before trying to start it

### PR DESCRIPTION
**Pull request by egabrovski-r3**
the same as https://github.com/corda/CSDE-cordapp-template-kotlin/pull/58 but using "development" as a base branch

### CODE REVIEW

File: buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java [hunk 1]
```diff
@@ -24,7 +24,11 @@ public CordaLifeCycleHelper(ProjectContext _pc) {
     }
 
     public void startCorda() throws IOException {
-        PrintStream pidStore = new PrintStream(new FileOutputStream(pc.cordaPidCache));
+        File cordaPIDFile = new File(pc.cordaPidCache);
+        if (cordaPIDFile.exists()) {
+            throw new IOException("Cannot start the Combined worker\nCached process ID file " + cordaPIDFile + " existing.\nWas the combined worker already started?");
```
Comment (yarivus) [inline buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java]: I prefer exception strings (explanations) not to contain `\n`

Also I think string interpolation is better than concat:
```suggestion
            throw new IOException("Cannot start the Combined worker. Cached process ID file ${cordaPIDFile} existing. Was the combined worker already started?");
```

**Review by yarivus — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java (hunk 1 above)
Comment (egabrovski-r3) [reply buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java]: I have used the same style and wording as in "stopCorda()". See line 94 in the same class. I can change it, but if so we should do it for both startCorda and stopCorda, if that's OK?

Comment (yarivus) [reply buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java]: I agree ... thanks :) 
@mattbradburyr3 FYI

Comment (egabrovski-r3) [reply buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java]: both messages are fixed
on "string interpolation is better than concat" this is Java code, and con cat here is not out of the box. Should I go with StringFormatter or we can leave it as it is?

Comment (yarivus) [reply buildSrc/src/main/java/com/r3/csde/CordaLifeCycleHelper.java]: No leave as is ;) 
Thanks

**Review by yarivus — APPROVED** [review]
LGTM

**Comment by egabrovski-r3** [comment]
qa-regpack-csde tests are OK

**Pull request merged**